src/taipy/core/data/mongo.py

```diff
@@ -0,0 +1,225 @@
+            raise InvalidExposedType(
+                f"Invalid string exposed type {exposed_type}. Supported values are {', '.join(self.__VALID_STRING_EXPOSED_TYPES)}"
+            )
+
+    @classmethod
+    def storage_type(cls) -> str:
+        return cls.__STORAGE_TYPE
+
+    @property  # type: ignore
+    @_self_reload(DataNode._MANAGER_NAME)
+    def encoder(self):
+        return self._encoder
+
+    @encoder.setter
+    def encoder(self, encoder: json.JSONEncoder):
+        self.properties[self._ENCODER_KEY] = encoder
+
+    @property  # type: ignore
+    @_self_reload(DataNode._MANAGER_NAME)
+    def decoder(self):
+        return self._decoder
+
+    @decoder.setter
+    def decoder(self, decoder: json.JSONDecoder):
+        self.properties[self._DECODER_KEY] = decoder
+
+    def _read(self):
+        if self.properties[self.__EXPOSED_TYPE_PROPERTY] == self.__EXPOSED_TYPE_PANDAS:
+            return self._read_as_pandas_dataframe()
+        if self.properties[self.__EXPOSED_TYPE_PROPERTY] == self.__EXPOSED_TYPE_NUMPY:
+            return self._read_as_numpy()
+        return self._read_as()
+
+    def _read_as(self):
+        custom_class = self.properties[self.__EXPOSED_TYPE_PROPERTY]
+        cursor = self._read_by_query()
+
+        return list(map(lambda row: custom_class(**row), cursor))
+
+    def _read_as_numpy(self):
+        return self._read_as_pandas_dataframe().to_numpy()
+
+    def _read_as_pandas_dataframe(self, columns: Optional[List[str]] = None):
+        cursor = self._read_by_query()
+        if columns:
+            return pd.DataFrame(list(cursor))[columns]
+        return pd.DataFrame(list(cursor))
+
+    def _read_by_query(self):
+        """Query from MongoDB, exclude the _id field"""
+
+        read_collection = self.mongo_db[self.properties.get("read_collection")]
+        query_result = list(read_collection.find(self.read_query, {"_id": 0}))
+
+        encoded_json = json.dumps(query_result)
+        return json.loads(encoded_json, cls=self._decoder)
+
+    def _write(self, data) -> None:
+        """Check data against a collection of types to handle insertion on the database."""
+
+        if isinstance(data, pd.DataFrame):
+            self._insert_dataframe(data)
+            return
+
+        if not isinstance(data, list):
+            data = [data]
+
+        if len(data) == 0:
+            read_collection = self.mongo_db[self.properties.get("read_collection")]
+            read_collection.delete_many({})
+            return
+
+        self._insert_dicts(self._encode_json(data))
+
+    def _insert_dicts(self, data: List[Dict]) -> None:
+        """
+        :param data: a list of dictionaries
+
+        This method will overwrite the data contained in a list of dictionaries into a collection.
+        """
+        write_collection = self.mongo_db[self.properties.get("write_collection")]
+        write_collection.delete_many({})
```

Comment:
Why do we have to delete_many before we insert new data?

Reply:
This is to overwrite old data.